### PR TITLE
Replace MD5 with SHA-1 in jenkins_plugins module.

### PR DESCRIPTION
**Pull request by karlmdavis**
##### SUMMARY

Fixes Issue #34304.

The `hashlib.md5(...)` function is disabled on systems running in FIPS 140-2 mode. To accommodate that, this commit just switches to a hash function (`hashlib.sha1(...)`) that is allowed by FIPS 140-2.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

`jenkins_plugin`

##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.4.1.0
  config file = None
  configured module search path = [u'/home/karl/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/karl/workspaces/cms/bluebutton-ansible-playbooks-data.git/venv/local/lib/python2.7/site-packages/ansible
  executable location = /home/karl/workspaces/cms/bluebutton-ansible-playbooks-data.git/venv/bin/ansible
  python version = 2.7.6 (default, Nov 23 2017, 15:49:48) [GCC 4.8.4]
```

But this PR is based off `devel`.

##### ADDITIONAL INFORMATION
Applied the same patch against `v2.4.1.0` and confirmed that it resolved the problem described in Issue #34304.

### CODE REVIEW

File: lib/ansible/modules/web_infrastructure/jenkins_plugin.py [hunk 1]
```diff
@@ -466,28 +470,36 @@ def install(self):
...
 
                     changed = True
                 else:
-                    # Get data for the MD5
+                    # Get data for the new SHA1
                     data = r.read()
 
                     # Make new checksum
-                    md5sum_new = hashlib.md5(data).hexdigest()
+                    try:
```
Comment (jtyr) [inline lib/ansible/modules/web_infrastructure/jenkins_plugin.py:490]: Do we need this exception here if we doing the checksum on a variable?

File: lib/ansible/modules/web_infrastructure/jenkins_plugin.py [hunk 2]
```diff
@@ -442,11 +442,15 @@ def install(self):
...
 
-            md5sum_old = None
-            if os.path.isfile(plugin_file):
-                # Make the checksum of the currently installed plugin
-                md5sum_old = hashlib.md5(
-                    open(plugin_file, 'rb').read()).hexdigest()
+            try:
+                sha1_old = hashlib.sha1(open(plugin_file, 'rb').read())
+            except Exception:
+                e = get_exception()
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/jenkins_plugin.py:448]: get_exception is deprecated API (used for Python 2.4). Could you please use native Python syntax for exception handling `except Exception as e`

File: lib/ansible/modules/web_infrastructure/jenkins_plugin.py [hunk 3]
```diff
@@ -442,11 +442,15 @@ def install(self):
...
+                self.module.fail_json(
+                    msg="Cannot calculate SHA1 of the old plugin.",
+                    details=e.message)
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/jenkins_plugin.py:451]: Use `to_native` for wrapping error message.

**Review by Akasurde — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@karlmdavis Are you still working on this PR ? 

needs_info

**Comment by karlmdavis** [comment]
:-|

Still _planning_ to... But I'm in the middle of a launch right now and haven't had a chance to get back to it. Definitely more than happy to address all the feedback once I do!

**Comment by Akasurde** [comment]
@karlmdavis Are you still working on this PR ?

needs_info

**Comment by jesstruck** [comment]
What are the plan of fixing this ??

**Pull request closed without merging**